src/rateEngine/__tests__/loadProfile.test.ts

```diff
@@ -128,7 +128,7 @@ describe('Load Profile', () => {
         const hourOfMonth = loadProfileOfOnes.expanded().find(({month, dayOfWeek}) => (
           month === monthIdx && dayOfWeek === 3
         )) 
-        return hourOfMonth.hourOfYear
+        return hourOfMonth?.hourOfYear ?? 0
```

Comment:
Is the default necessary here? the type looks to be a required number:

https://github.com/bellawatt/electric-rate-engine/blob/f741cdfbba045e058e2d2359935d3c2bfdc724e9/src/rateEngine/types/index.ts#L211

Reply:
Yeah I changed this test to use findIndex. If you look further down there's this:

```maxhours.forEach((hour, idx) => {
  loadProfileData[hour] = maxes[idx];
});```

loadProfileData is type `number[]` and maxhours is `DetailedLoadProfileHour[]`. This means (when using .find) `hour` here would be an object and using an object to set an array index is surely a bug.

Reply:
how was this ever passing then?

Reply:
just realized I chopped off the `.hourOfYear` so that's what the issue is. Reverted the test to use `.find` (with a default to satisfy typescript) and a comment